apps/laboratory/src/components/Wagmi/WagmiGetCallsStatusTest.tsx

```diff
@@ -1,86 +1,71 @@
 import { Button, Stack, Text, Input } from '@chakra-ui/react'
-import { useAccount, useConnections } from 'wagmi'
-import { EthereumProvider } from '@walletconnect/ethereum-provider'
-import { useState, useEffect } from 'react'
+import { useAccount } from 'wagmi'
+import { useState, useCallback } from 'react'
 import { useChakraToast } from '../Toast'
 import { EIP_5792_RPC_METHODS } from '../../utils/EIP5792Utils'
 import { bigIntReplacer } from '../../utils/CommonUtils'
 import { useCallsStatus } from 'wagmi/experimental'
+import { useWagmiAvailableCapabilities } from '../../hooks/useWagmiActiveCapabilities'
 
 export function WagmiGetCallsStatusTest() {
-  const [ethereumProvider, setEthereumProvider] =
-    useState<Awaited<ReturnType<(typeof EthereumProvider)['init']>>>()
-  const [enableGetCallsStatus, setEnableGetCallsStatus] = useState(false)
-  const [batchCallId, setBatchCallId] = useState('')
+  const { ethereumProvider, isMethodSupported: isGetCallsStatusSupported } =
+    useWagmiAvailableCapabilities({
+      method: EIP_5792_RPC_METHODS.WALLET_GET_CALLS_STATUS
+    })
 
   const { status, address } = useAccount()
+
   const isConnected = status === 'connected'
-  const connection = useConnections()
+
+  if (!isConnected || !ethereumProvider || !address) {
+    return (
+      <Text fontSize="md" color="yellow">
+        Wallet not connected
+      </Text>
+    )
+  }
+
+  if (!isGetCallsStatusSupported()) {
+    return (
+      <Text fontSize="md" color="yellow">
+        Wallet does not support wallet_getCallsStatus rpc method
+      </Text>
+    )
+  }
+
+  return <AvailableTestContent />
+}
+
+function AvailableTestContent() {
+  const [batchCallId, setBatchCallId] = useState('')
+
   const toast = useChakraToast()
-  const {
-    isLoading,
-    isFetched,
-    error,
-    data: callsStatusResult
-  } = useCallsStatus({
+  const { isLoading, refetch: getCallsStatus } = useCallsStatus({
     id: batchCallId,
     query: {
-      enabled: enableGetCallsStatus && isConnected
+      enabled: false
```

Comment:
is this ok ?  cc @KannuSingh

Reply:
The idea is to disable the fetch on render and be triggered only by the `refetch` function

Reply:
Looks good.

Reply:
Can you just check if this not getting triggered on empty batchId?